### PR TITLE
Fix gcylc filter dialog bug (introduced since last release)

### PR DESCRIPTION
**Pull request by hjoliver**
Bug introduced by 75ba8a6c

### CODE REVIEW

File: lib/cylc/gui/app_gcylc.py [hunk 1]
```diff
@@ -3077,8 +3077,8 @@ def check_task_filter_buttons(self, tb=None):
                 box = ebox.get_children()[0]
                 try:
                     cb_ = box.get_children()[1]
-                except (ValueError, AttributeError):
-                    # ValueError: an empty box to line things up.
+                except (IndexError, AttributeError):
+                    # IndexError: an empty box to line things up.
                     # AttributeError: the name filter entry box.
```
Comment (hjoliver) [inline lib/cylc/gui/app_gcylc.py]:
```
empty = []
_, a = empty  # AttributeError
a = empty[1]  # ValueError
```

TODO - we've made a bunch of these changes recently, to get rid of spurious variables. Better go back and check for corresponding changes to exceptions...

Comment (oliver-sanders) [reply lib/cylc/gui/app_gcylc.py]: Nasty, I'll see if I can find any more examples...

Comment (oliver-sanders) [reply lib/cylc/gui/app_gcylc.py]: I can't see any in #2318 or #2375.

Comment (hjoliver) [reply lib/cylc/gui/app_gcylc.py]: Thanks for checking!

**Pull request merged**